test/test_gpu_tensorcore.cpp

```diff
@@ -3991,6 +3992,14 @@ TEST_F(NVFuserTest, FusionAmpereMatmulSmemEpilogue_CUDA) {
       GTEST_SKIP()
           << "Test conducted without utilizing shared memory epilogue due to the device's constrained shared memory capacity.";
     }
+
+    auto smem_allocs = fe.kernel()->summary().dynamic_smem_allocations;
+    TORCH_CHECK(smem_allocs.size() == 3);
+    if (params.promote_prologue_smem_reuse) {
+      // Check prologue shared memory re-use
+      TORCH_CHECK(smem_allocs.at(1)->address()->isZero());
+      TORCH_CHECK(smem_allocs.at(2)->address()->isZero());
```

Comment:
Is `smem_allocs.at(0)` A and `smem_allocs.at(1)` B, and `smem_allocs.at(1)` C? So B is reusing A's memory, and C is reusing B's memory?

Reply:
Yes that's right. I improved the comment here a bit.

Reply:
I see. So we allocate B first, then A. This makes sense. Thanks for the explanation!

Reply:
Wait, is it guaranteed that which of A vs B is allocated first? If A is allocated first, then will `smem_allocs.at(1)` no longer be zero here? I think we should remove the check for `smem_allocs.at(1)`.

Reply:
Very good question since their lifetimes end at the same time point. We break ties like that by ordering by `name()`, so `B` will be ordered after `A`, leading to this order consistently. However, that's not exactly clear so maybe we could just check that `C` is placed at `0` and that either `A` or `B` is at 0.